vllm/model_executor/models/internvl.py

```diff
@@ -449,11 +449,12 @@ def _parse_and_validate_image_input(
             if not isinstance(pixel_values, (torch.Tensor, list)):
                 raise ValueError("Incorrect type of pixel values. "
                                  f"Got type: {type(pixel_values)}")
-
+            # We need to flatten (B, N, P) to (B*N*P),
+            # so we call flatten_bn twice.
             return InternVLImagePixelInputs(
```

Comment:
Can you update the docstrings in `InternVLImagePixelInputs` to use the new shape?

Reply:
The docstrings in `InternVLImagePixelInputs` should have been in the correct shape, where `batch_size * num_images * (1 + num_patches)` is same as `(B*N*P)`:
https://github.com/vllm-project/vllm/blob/1bf2dd9df025feb82e27f90f534a3bf829ae75e9/vllm/model_executor/models/internvl.py#L47-L53

Reply:
Oh, so the actual shape was wrong before this PR. Got it!

Reply:
In fact, the root issue is the order of `flatten(B, N, P)` instead of the shape. We use `concat=True` at first flatten before (concat on `N` dim), which can't work when images have various `num_patches`. And we can use `concat=True` at last flatten (concat on `P` dim) to prevent this conflicts.